nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -246,6 +271,32 @@ void BtActionServer<ActionT>::executeCallback()
   }
 }
 
+template<class ActionT>
+void BtActionServer<ActionT>::populateErrorCode(
+  typename std::shared_ptr<typename ActionT::Result> result)
+{
+  int highest_priority_error_code = std::numeric_limits<int>::max();
+  for (const auto & error_code_id : error_code_ids_) {
+    // TODO(jwallace42): Using try catch due to
```

Comment:
TODO?

Reply:
Once // https://github.com/BehaviorTree/BehaviorTree.CPP/issues/271 gets fixed the try catch can be removed.

Reply:
OK - but you have a TODO, I don't allow TODOs to be merged. If its to-do, it should be do-ed :laughing:

Reply:
Makes sense. 

I was following the old pattern :). 

https://github.com/ros-planning/navigation2/blob/main/nav2_navfn_planner/src/navfn_planner.cpp#L231

https://github.com/ros-planning/navigation2/blob/main/nav2_smac_planner/test/deprecated/upsampler_cost_function.hpp#L251

Reply:
Well I hope 97 of them are before my time. Those in the constrained smoother are in a depreciated file (see the directory) and not meant for production. That's there for historical reference and was never completed